hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -946,7 +946,7 @@ public AclStatus getAclStatus(final Path path) throws IOException {
   @Override
   public void access(final Path path, final FsAction mode) throws IOException {
     LOG.debug("AzureBlobFileSystem.access path : {}, mode : {}", path, mode);
-    Path qualifiedPath = makeQualified(path);
+    Path qualifiedPath = path == null ? null : makeQualified(path);
```

Comment:
The original line 949 is correct.  If path is null, FileSystem.access should throw IllegalArgumentException and that's what calling makeQualified(path) will do.

Reply:
Made it this way, otherwise even if the flag for the check access is false this will throw IllegalArgumentException. Which will affect backward compatibility.

Reply:
Keeping it this way as we don't need to break the existing flow when the checkaccess flag is false.

Reply:
I don’t follow.  What flow are you referring to?  I think you should revert this line because null path should result in IllegalArgumentException. Calling makeQualified is a standard thing done by all the APIs taking a Path.

Reply:
Removing the addition as discussed.